### PR TITLE
Fix/adjust the created at format of port phone service insert event command member note to date

### PR DESCRIPTION
**Pull request by LoisChen68**

### CODE REVIEW

**Comment by LoisChen68** [comment]
[(1)【聯絡紀錄】porter搬移工掛掉這情況下不能讓member_note的建立時間跑掉](https://trello.com/c/d0PVjoyx/5517-1%E3%80%90%E8%81%AF%E7%B5%A1%E7%B4%80%E9%8C%84%E3%80%91porter%E6%90%AC%E7%A7%BB%E5%B7%A5%E6%8E%9B%E6%8E%89%E9%80%99%E6%83%85%E6%B3%81%E4%B8%8B%E4%B8%8D%E8%83%BD%E8%AE%93membernote%E7%9A%84%E5%BB%BA%E7%AB%8B%E6%99%82%E9%96%93%E8%B7%91%E6%8E%89)

File: src/runner/porter-command/portPhoneServiceInsertEventCommand.ts [hunk 1]
```diff
@@ -100,7 +101,11 @@ class PortPhoneServiceInsertEventCommand implements PorterCommand {
           async (data) =>
             await manager.transaction(async (manager) => {
               try {
-                const { memberNotes, lastMemberNotes, key } = data;
+                const { lastMemberNotes, key } = data;
+                const memberNotes = data.memberNotes.map((note) => ({
+                  ...note,
+                  createdAt: dayjs(note.createdAt).toDate(),
+                }));
```
Comment (TedxTed) [inline src/runner/porter-command/portPhoneServiceInsertEventCommand.ts]: ORM best practice is use ORM instance 

```ts
  const memberNotes = data.memberNotes.map((note) => {
    const memberNote = new MemberNote();
    memberNote.authorId = note?.authorId || '';
    memberNote.metadata = note?.metadata || null;
    memberNote.duration = note?.duration || 0;
    memberNote.memberId = note?.memberId || '';
    memberNote.status = note?.status ;
    memberNote.type = note?.type ; 
    memberNote.createdAt = dayjs(note?.createdAt).isValid() ? dayjs(note.createdAt).toDate() : new Date();
    return memberNote;
  });
  
// need to update insert function too
  async insertData(data: MemberNote[], entityManager: EntityManager) {
    const memberNoteRepo = entityManager.getRepository(MemberNote);
    return await memberNoteRepo.insert(data);
  }
```

**Review by TedxTed — CHANGES_REQUESTED** [review]

File: src/runner/porter-command/portPhoneServiceInsertEventCommand.ts (hunk 1 above)
Comment (LoisChen68) [reply src/runner/porter-command/portPhoneServiceInsertEventCommand.ts]: Adjusted, thanks for review.

File: src/runner/porter-command/portPhoneServiceInsertEventCommand.ts [hunk 2]
```diff
@@ -100,15 +101,28 @@ class PortPhoneServiceInsertEventCommand implements PorterCommand {
...
+                const memberNotes = data.memberNotes
+                  ? data.memberNotes.map((note) => {
+                      const memberNote = new MemberNote();
+                      memberNote.authorId = note?.authorId || '';
+                      memberNote.metadata = note?.metadata || null;
+                      memberNote.duration = note?.duration || 0;
+                      memberNote.memberId = note?.memberId || '';
+                      memberNote.status = note?.status;
+                      memberNote.type = note?.type;
+                      memberNote.createdAt = note?.createdAt ? dayjs(note.createdAt).toDate() : new Date();
```
Comment (TedxTed) [inline src/runner/porter-command/portPhoneServiceInsertEventCommand.ts]: `note.createdAt`
I am concerned that the source createdAt might be in a corrupted or unrecognizable format
maybe need to use .isValid() function judge time format correctly

File: src/runner/porter-command/portPhoneServiceInsertEventCommand.ts (hunk 1 above)
Comment (LoisChen68) [reply src/runner/porter-command/portPhoneServiceInsertEventCommand.ts]: Note:
After some discussion, in addition to checking createdAt here, we also need to check and convert the data source in kolable-server. For details, see: https://github.com/urfit-tech/kolable-server/pull/103/files

**Review by TedxTed — APPROVED** [review]

**Pull request merged**